src/transfers.rs

```diff
@@ -216,313 +317,30 @@ fn do_generates_transfers(
-                comment_links: CommentLinksT::default(),
-                visible: true,
-                coord: Coord {
-                    lon: 2.369517087936402,
-                    lat: 48.845301913401144,
-                },
-                timezone: None,
-                geometry_id: None,
-                equipment_id: None,
-                stop_area_id: "sa_1".to_string(),
-                fare_zone_id: None,
-                stop_type: StopType::Point,
-            },
-        ]).unwrap();
-
-        let rules = vec![
-            Rule {
-                from_stop_id: "sp_3".to_string(),
-                to_stop_id: "sp_3".to_string(),
-                transfer_time: None,
-            },
-            Rule {
-                from_stop_id: "sp_2".to_string(),
-                to_stop_id: "sp_3".to_string(),
-                transfer_time: Some(100),
-            },
-        ];
-
-        let transfers =
-            super::do_generates_transfers(&mut transfers, &stop_points, 100.0, 0.785, 120, &rules);
-
-        assert_eq!(
-            transfers,
-            vec![
-                Transfer {
-                    from_stop_id: "sp_1".to_string(),
-                    to_stop_id: "sp_1".to_string(),
-                    min_transfer_time: Some(0),
-                    real_min_transfer_time: Some(120),
-                    equipment_id: None
-                },
-                Transfer {
-                    from_stop_id: "sp_1".to_string(),
-                    to_stop_id: "sp_2".to_string(),
-                    min_transfer_time: Some(50),
-                    real_min_transfer_time: Some(60),
-                    equipment_id: None
-                },
-                Transfer {
-                    from_stop_id: "sp_1".to_string(),
-                    to_stop_id: "sp_3".to_string(),
-                    min_transfer_time: Some(200),
-                    real_min_transfer_time: Some(210),
-                    equipment_id: None
-                },
-                Transfer {
-                    from_stop_id: "sp_2".to_string(),
-                    to_stop_id: "sp_1".to_string(),
-                    min_transfer_time: Some(83),
-                    real_min_transfer_time: Some(203),
-                    equipment_id: None
-                },
-                Transfer {
-                    from_stop_id: "sp_2".to_string(),
-                    to_stop_id: "sp_2".to_string(),
-                    min_transfer_time: Some(0),
-                    real_min_transfer_time: Some(120),
-                    equipment_id: None
-                },
-                Transfer {
-                    from_stop_id: "sp_2".to_string(),
-                    to_stop_id: "sp_3".to_string(),
-                    min_transfer_time: Some(100),
-                    real_min_transfer_time: Some(220),
-                    equipment_id: None
-                }
-            ]
-        );
+    model.transfers = Collection::new(new_transfers);
```

Comment:
This can't be done without `<Model as DerefMut>`

Reply:
yes that's the problem, do you have an idea to do the trick?

Reply:
implement an `into_collections(self) -> Collections` on `Model`.